panther_manager/include/panther_manager/shutdown_host_node.hpp

```diff
@@ -0,0 +1,109 @@
+#ifndef PANTHER_MANAGER_SHUTDOWN_HOST_NODE_HPP_
+#define PANTHER_MANAGER_SHUTDOWN_HOST_NODE_HPP_
+
+#include <behaviortree_cpp/action_node.h>
+#include <behaviortree_cpp/bt_factory.h>
+#include <libssh/libssh.h>
+
+#include <ros/ros.h>
+
+namespace panther_manager
+{
+
+class ShutdownHost : public BT::StatefulActionNode
+{
+public:
+  ShutdownHost(const std::string & name, const BT::NodeConfig & conf)
+  : BT::StatefulActionNode(name, conf)
+  {
+    node_name_ = ros::this_node::getName();
+  }
+
+  ssh_session session_;
+  ssh_channel channel_;
+
+  bool check_ip(const std::string ip)
+  {
+    return system(("ping -c 1 -w 1 " + ip + " > /dev/null").c_str()) == 0;
+  }
+
+  int ssh_execute_command(const char * host, const char * user, const char * command)
+  {
+    ROS_INFO("[%s] Shutting down device at: %s", node_name_.c_str(), host);
+
+    session_ = ssh_new();
+    if (session_ == NULL) return -1;
+
+    ssh_options_set(session_, SSH_OPTIONS_HOST, host);
+    ssh_options_set(session_, SSH_OPTIONS_USER, user);
+    ssh_options_set(session_, SSH_OPTIONS_PORT, &port_);
+    ssh_options_set(session_, SSH_OPTIONS_LOG_VERBOSITY, &verbosity_);
+
+    int rc = ssh_connect(session_);
+    if (rc != SSH_OK) {
+      ROS_ERROR("[%s] Error connecting to host: %s", node_name_.c_str(), ssh_get_error(session_));
+      ssh_free(session_);
+      return -1;
+    }
+
+    rc = ssh_userauth_publickey_auto(session_, NULL, NULL);
+    if (rc != SSH_AUTH_SUCCESS) {
+      ROS_ERROR(
+        "[%s] Error authenticating with public key: %s", node_name_.c_str(),
+        ssh_get_error(session_));
+      ssh_disconnect(session_);
+      ssh_free(session_);
+      return -1;
+    }
+
+    channel_ = ssh_channel_new(session_);
+    if (channel_ == NULL) {
+      ROS_ERROR(
+        "[%s] Failed to create ssh channel: %s", node_name_.c_str(), ssh_get_error(session_));
+      ssh_disconnect(session_);
+      ssh_free(session_);
+      return -1;
+    }
+
+    rc = ssh_channel_open_session(channel_);
+    if (rc != SSH_OK) {
+      ROS_ERROR("[%s] Failed to open ssh channel: %s", node_name_.c_str(), ssh_get_error(session_));
+      ssh_channel_free(channel_);
+      ssh_disconnect(session_);
+      ssh_free(session_);
+      return -1;
+    }
+
+    rc = ssh_channel_request_exec(channel_, command);
+    if (rc != SSH_OK) {
+      ROS_ERROR(
+        "[%s] Failed to execute ssh command: %s", node_name_.c_str(), ssh_get_error(session_));
+      ssh_channel_close(channel_);
+      ssh_channel_free(channel_);
+      ssh_disconnect(session_);
+      ssh_free(session_);
+      return -1;
+    }
+
+    return 0;
+  }
+
+  void close_connection(){
+    ssh_channel_send_eof(channel_);
+    ssh_channel_close(channel_);
+    ssh_channel_free(channel_);
+    ssh_disconnect(session_);
+    ssh_free(session_);
+  }
+
+  std::string get_node_name() { return node_name_; }
+
+private:
+  int verbosity_ = SSH_LOG_NOLOG;
+  int port_ = 22;
```

Comment:
```suggestion
  int port_ = 22;
```
This could be a bt port with default set to 22

Reply:
Shutdowon hosts will be separate PR. Can be skipped by now